lms/services/email_unsubscribe.py

```diff
@@ -13,14 +13,27 @@ def __init__(self, db, jwt_service: JWTService, secret: str, route_url: Callable
         self._secret = secret
         self._route_url = route_url
 
+    def settings_url(self, h_userid):
+        """Return a URL for the email settings page for the given h_userid.
+
+        The URL will contain an authentication token for h_userid in a query
+        param.
+        """
+        return self._route_url(
+            "email.settings",
+            _query={
+                "token": self._generate_token(h_userid, lifetime=timedelta(minutes=30))
```

Comment:
I reckon in general the lifetime is not that important (within reason), this being correctly scoped to the settings page is key and that's seem to be handled with the cookie and the policy.

I mean, it would be very difficult for us to make a mistake and starting this tokens as auth for something else :+1: